packages/@glimmer/runtime/lib/compiled/opcodes/component.ts

```diff
@@ -77,13 +77,18 @@ APPEND_OPCODES.add(Op.PrepareArgs, (vm, { op1: _state }) => {
 
     let names = Object.keys(named);
     let namedCount = names.length;
+    let atNames = [];
 
     for (let i = 0; i < namedCount; i++) {
-      stack.push(named[names[i]]);
+      let value = named[names[i]];
+      let atName = `@${names[i]}`;
+
+      stack.push(value);
+      atNames.push(atName);
     }
 
-    stack.push(names);
-    args.setup(stack, true);
+    stack.push(atNames);
```

Comment:
This is basically how the args passing convention works. Downstream things expect the stack to include this array of names (so that it can pop all the named / positional references and know what they belong to). 

Some (likely better) inline docs here: https://github.com/glimmerjs/glimmer-vm/blob/master/packages/@glimmer/runtime/lib/vm/arguments.ts#L11-L14